src/detect-engine-content-inspection.c

```diff
@@ -541,20 +543,18 @@ uint8_t DetectEngineContentInspection(DetectEngineCtx *de_ctx, DetectEngineThrea
 
         /* if we have dce enabled we will have to use the endianness
          * specified by the dce header */
-        if ((bmd->flags & DETECT_BYTEMATH_FLAG_ENDIAN) &&
-            endian == DETECT_BYTEMATH_ENDIAN_DCE &&
-            flags & (DETECT_CI_FLAGS_DCE_LE|DETECT_CI_FLAGS_DCE_BE)) {
+        if ((bmd->flags & DETECT_BYTEMATH_FLAG_ENDIAN) && endian == (int)EndianDCE &&
+                flags & (DETECT_CI_FLAGS_DCE_LE | DETECT_CI_FLAGS_DCE_BE)) {
 
             /* enable the endianness flag temporarily.  once we are done
              * processing we reset the flags to the original value*/
-            endian |= ((flags & DETECT_CI_FLAGS_DCE_LE) ?
-                       DETECT_BYTEMATH_ENDIAN_LITTLE : DETECT_BYTEMATH_ENDIAN_BIG);
+            endian |= (int)((flags & DETECT_CI_FLAGS_DCE_LE) ? LittleEndian : BigEndian);
```

Comment:
This cast looks strange to me.
Do you get a compiler warning without it ?

Reply:
No, no compiler warnings. But,  the cast should be to a `uint8_t` instead of an `int`. I'll update.

Reply:
Ok, makes more sense to me then :-)